File: app/src/main/java/com/example/grocer21/MyDietRecyclerViewAdapter.java
package com.example.grocer21;

import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.grocer21.Database.Diet;
import com.example.grocer21.DietFragment.OnListFragmentInteractionListener;

import java.util.List;
import java.util.Objects;

/**
 * {@link RecyclerView.Adapter} that can display a {@link Diet} and makes a call to the
 * specified {@link OnListFragmentInteractionListener}.
 */
public class MyDietRecyclerViewAdapter extends RecyclerView.Adapter<MyDietRecyclerViewAdapter.ViewHolder> {

    private final LiveData<List<Diet>> dietList;
    private final OnListFragmentInteractionListener mListener;

    MyDietRecyclerViewAdapter(LiveData<List<Diet>> items, OnListFragmentInteractionListener listener) {
        dietList = items;
        mListener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.fragment_diet, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        holder.mItem = Objects.requireNonNull(dietList.getValue()).get(position);
        holder.mIdView.setText(dietList.getValue().get(position).getName());

        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (null != mListener) {
                    // Notify the active callbacks interface (the activity, if the
                    // fragment is attached to one) that an item has been selected.
                    mListener.onListFragmentInteraction(holder.mItem);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return Objects.requireNonNull(dietList.getValue()).size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        final View mView;
        final TextView mIdView;
        Diet mItem;

        ViewHolder(View view) {
            super(view);
            mView = view;
            mIdView = view.findViewById(R.id.item_number);
        }

        @NonNull
        @Override
        public String toString() {
            return super.toString() + " '" + "'";
        }
    }
}

File: app/src/main/java/com/example/grocer21/MyDietRecyclerViewAdapter.kt
package com.example.grocer21

import android.arch.lifecycle.LiveData
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.example.grocer21.DietFragment.OnListFragmentInteractionListener
import com.example.grocer21.database.entities.Diets
import java.util.Objects

/**
 * [RecyclerView.Adapter] that can display a [Diets] and makes a call to the
 * specified [OnListFragmentInteractionListener].
 */
class MyDietRecyclerViewAdapter internal constructor(private val dietList: LiveData<List<Diets>>, private val mListener: OnListFragmentInteractionListener?) : RecyclerView.Adapter<MyDietRecyclerViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.fragment_diet, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.mItem = Objects.requireNonNull<List<Diets>>(dietList.value)[position]
        holder.mIdView.text = dietList.value!![position].getName()

        holder.mView.setOnClickListener {
            mListener?.onListFragmentInteraction(holder.mItem!!)
        }
    }

    override fun getItemCount(): Int {
        return Objects.requireNonNull<List<Diets>>(dietList.value).size
    }

    inner class ViewHolder internal constructor(internal val mView: View) : RecyclerView.ViewHolder(mView) {
        internal val mIdView: TextView = mView.findViewById(R.id.item_number)
        internal var mItem: Diets? = null

        override fun toString(): String {
            return super.toString() + " '" + "'"
        }
    }
}
